go_modules/lib/dependabot/go_modules/file_parser.rb

```diff
@@ -31,8 +33,40 @@ def parse
         dependency_set.dependencies
       end
 
+      sig { returns(Ecosystem) }
+      def ecosystem
+        @ecosystem ||= T.let(
+          Ecosystem.new(
+            name: ECOSYSTEM,
+            package_manager: package_manager,
+            language: language
+          ),
+          T.nilable(Ecosystem)
+        )
+      end
+
       private
 
+      sig { returns(Ecosystem::VersionManager) }
+      def package_manager
+        @package_manager ||= T.let(
+          PackageManager.new(go_version),
+          T.nilable(Dependabot::GoModules::PackageManager)
+        )
+      end
+
+      sig { returns(T.nilable(Ecosystem::VersionManager)) }
+      def language
+        @language ||= T.let(begin
+          Language.new(go_version)
+        end, T.nilable(Dependabot::GoModules::Language))
+      end
```

Comment:
These both appear to do the same thing, which doesn't seem very useful. 

Go doesn't have a separate package manager, it uses the toolchain version either from the `go` directive or the optional `toolchain` directive. Both of those would be useful to gather. You already have the first with `language` here, so I'd switch the `package_manager` to `toolchain` and figure out how to gather that, if present.

Reply:
Ok, I'll take a look at how I can get the toolchain

Reply:
This is done